### PR TITLE
[stable/prometheus] Adding Vertical Pod Autoscaler support

### PR DESCRIPTION
**Pull request by ali-sattari**
## What this PR does / why we need it:
Adds support to define Vertical Pod Autoscaling for Prometheus pod to the chart.

#### Special notes for your reviewer:
VPA is still in beta, but many started using it. It is possible to activate it for `server` container and all other containers (sidecars basically) through `containerPolicies`.

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by homelessnessbo** [comment]
This would be really nice to have!

**Comment by zanhsieh** [comment]
@mgoodness
@gianrubio
Could you approve this please? Thank you.

**Comment by Vlaaaaaaad** [comment]
This might be a great help for https://github.com/prometheus/prometheus/issues/5560 so bumping this. 

Are there any concerns with this PR?

**Comment by Vlaaaaaaad** [comment]
I am still very much for this so commenting to get rid of the stale label. 

This is stuck due to the Prometheus chart not having owners to review and merge PRs.

**Comment by homelessnessbo** [comment]
PLEASE MERGE PLEASE MERGE WE USE IT WE USE IT

**Comment by homelessnessbo** [comment]
@ali-sattari update Chart.yaml

**Comment by zanhsieh** [comment]
@ali-sattari 
Please bump the chart to latest. Sorry for long waiting.
/assign

**Comment by ali-sattari** [comment]
@zanhsieh done.

**Review by zanhsieh — APPROVED** [review]

File: stable/prometheus/templates/server-vpa.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    apiVersion: "apps/v1"
+    kind: StatefulSet
+{{- else }}
+    apiVersion: "extensions/v1beta1"
+    kind: Deployment
+{{- end }}
+    name: {{ template "prometheus.server.fullname" . }}
+  updatePolicy:
+    updateMode: {{ .Values.server.verticalAutoscaler.updateMode | default "Off" | quote }}
+  resourcePolicy:
```
Comment (nishant-ketu) [inline stable/prometheus/templates/server-vpa.yaml]: getting a error on this line. "Error: YAML parse error on prometheus/templates/server-vpa.yaml: error converting YAML to JSON: yaml: line 21: could not find expected ':'"

**Pull request merged**